scripts/gen-readme.go

```diff
@@ -0,0 +1,252 @@
+type ReadmeTemplateConfig struct {
+	Title   string
+	Actions []ReadmeAction
+}
+
+// Workflow is the object properties for each workflow
+type Workflow struct {
+	Starter        bool   `json:"starter"`
+	Type           string `json:"type"`
+	WorkflowPath   string `json:"workflowPath"`
+	PropertiesPath string `json:"propertiesPath"`
+}
+
+// WorkflowConfig is the object referencing all workflow configs
+type WorkflowConfig map[string]Workflow
+
+func main() {
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+
+	if err := realMain(ctx); err != nil {
+		cancel()
+		fmt.Fprintf(os.Stderr, "%s\n", err)
+		os.Exit(1)
+	}
+}
+
+func realMain(ctx context.Context) error {
+	configBytes, err := os.ReadFile(workflowConfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to read workflow config file: %w", err)
+	}
+
+	var workflowConfig WorkflowConfig
+	if err := json.Unmarshal(configBytes, &workflowConfig); err != nil {
+		return fmt.Errorf("failed to unmarshal workflow config: %w", err)
+	}
+
+	hasInvalidConfigs := false
+
+	sortedWorkflowsIDs := getSortedWorkflowIDs(workflowConfig)
+
+	readmeActions := map[string]ReadmeAction{}
+
+	for _, workflowID := range sortedWorkflowsIDs {
+		workflow := workflowConfig[workflowID]
+
+		if _, err := os.Stat(workflow.WorkflowPath); os.IsNotExist(err) {
+			hasInvalidConfigs = true
+			fmt.Println(fmt.Sprintf("workflow file does not exist for workflow %s: path - %s", workflowID, workflow.WorkflowPath))
+		}
+
+		propertiesBytes, err := os.ReadFile(workflow.PropertiesPath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				hasInvalidConfigs = true
+				fmt.Println(fmt.Sprintf("properties file does not exist for workflow %s: path - %s", workflowID, workflow.PropertiesPath))
+			} else {
+				return fmt.Errorf("failed to read properties file %s: %w", workflow.PropertiesPath, err)
+			}
+		}
+
+		if hasInvalidConfigs {
+			continue
+		}
+
+		var propertiesConfigs PropertiesConfig
+		if err := json.Unmarshal(propertiesBytes, &propertiesConfigs); err != nil {
+			return fmt.Errorf("failed to unmarshal properties file %s: %w", workflow.PropertiesPath, err)
+		}
+
+		parts := strings.Split(workflow.WorkflowPath, "/")
+
+		if len(parts) < 2 {
+			return errors.New("invalid workflow path %s, path should have at least 3 folders")
+		}
+
+		actionName := parts[1]
+		actionPath := path.Join(parts[:2]...)
+		actionReadMePath := path.Join(actionPath, "README.md")
+
+		if _, err := os.Stat(actionReadMePath); err != nil {
+			if os.IsNotExist(err) {
+				actionReadMeContents := fmt.Sprintf("# %s examples", actionName)
+				if err := os.WriteFile(actionReadMePath, []byte(actionReadMeContents), 0644); err != nil {
+					return fmt.Errorf("failed writing content to action README file %s: %w", actionReadMePath, err)
+				}
+			} else {
+				return fmt.Errorf("failed to validate %s README file exists: %w", actionReadMePath, err)
+			}
```

Comment:
This seems like something we should just throw an error on and ask users to generate if it doesnt exist via new-workflow?

Reply:
my thought here was that something "else" went wrong so inform the user what happened

Reply:
Right but I was thinking if we should just complain for any err rather than handling  the case where it doesnt exist and regenerating an empty readme.

Reply:
ah i hear ya, let me adjust, i think that makes sense because this readme SHOULD exist already